### PR TITLE
Include cxxbridge-cmd in Cargo.lock, check version consistency

### PR DESCRIPTION
**Pull request by djmitche**
This adds cxxbridge-cmd to Cargo.lock per https://github.com/dtolnay/cxx/issues/1407#issuecomment-2509136343, as linked from #3705.

It adds an MSRV to `src/taskchampion-cpp/Cargo.toml` so that the version of `Cargo.lock` is stil compatible with the MSRV.

It additionally adds a check of the Cargo metadata for all of the cxx* versions agreeing, and for the MSRVs agreeing.

cc: @doronbehar

### CODE REVIEW

File: .github/workflows/metadata-check.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  ok = false
+fi
+if [ "${cxx_version}" != "${cxxbridge_flags_version}" ]; then
+  echo "Found differing cxxbridge-flags version ${cxxbridge_flags_version}"
+  ok = false
+fi
+if [ "${cxx_version}" != "${cxxbridge_macro_version}" ]; then
+  echo "Found differing cxxbridge-macro version ${cxxbridge_macro_version}"
+  ok = false
+fi
```
Comment (doronbehar) [inline .github/workflows/metadata-check.sh]: For loop?

**Review by doronbehar — COMMENTED** [review]
Looks pretty good! Thanks for making the effort with that GitHub workflow.

**Comment by doronbehar** [comment]
Haven't tested this locally yet.

File: .github/workflows/metadata-check.sh (hunk 1 above)
Comment (djmitche) [reply .github/workflows/metadata-check.sh]: Meh, maybe if we add more :)

File: .github/workflows/metadata-check.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+get_msrv() {
+  local package="${1}"
+  jq -r '.packages[] | select(.name == "'"${package}"'") | .rust_version' "${META}"
+}
+
+# check that the cxx packages all have the same version
+check_cxx_versions() {
+  local cxx_version=$(get_version "cxx")
+  local cxx_build_version=$(get_version "cxx-build")
+  local cxxbridge_cmd_version=$(get_version "cxx-build")
```
Comment (dtolnay) [inline .github/workflows/metadata-check.sh:26]:
```suggestion
  local cxxbridge_cmd_version=$(get_version "cxxbridge-cmd")
```

File: .github/workflows/metadata-check.sh [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  fi
+  if [ "${cxx_version}" != "${cxxbridge_cmd_version}" ]; then
+    echo "Found differing cxxbridge-cmd version ${cxxbridge_cmd_version}"
+    ok = false
+  fi
+  if [ "${cxx_version}" != "${cxxbridge_flags_version}" ]; then
+    echo "Found differing cxxbridge-flags version ${cxxbridge_flags_version}"
+    ok = false
+  fi
+  if [ "${cxx_version}" != "${cxxbridge_macro_version}" ]; then
```
Comment (dtolnay) [inline .github/workflows/metadata-check.sh:44]: This is already enforced by https://github.com/dtolnay/cxx/blob/1.0.124/Cargo.toml#L26.

File: .github/workflows/metadata-check.sh [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  echo "Found cxx version ${cxx_version}"
+  if [ "${cxx_version}" != "${cxx_build_version}" ]; then
+    echo "Found differing cxx-build version ${cxx_build_version}"
+    ok = false
+  fi
+  if [ "${cxx_version}" != "${cxxbridge_cmd_version}" ]; then
+    echo "Found differing cxxbridge-cmd version ${cxxbridge_cmd_version}"
+    ok = false
+  fi
+  if [ "${cxx_version}" != "${cxxbridge_flags_version}" ]; then
```
Comment (dtolnay) [inline .github/workflows/metadata-check.sh:40]: This is already enforced by https://github.com/dtolnay/cxx/blob/1.0.124/Cargo.toml#L31.

File: .github/workflows/metadata-check.sh [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+check_cxx_versions() {
+  local cxx_version=$(get_version "cxx")
+  local cxx_build_version=$(get_version "cxx-build")
+  local cxxbridge_cmd_version=$(get_version "cxx-build")
+  local cxxbridge_flags_version=$(get_version "cxxbridge-flags")
+  local cxxbridge_macro_version=$(get_version "cxxbridge-macro")
+
+  ok=true
+  echo "Found cxx version ${cxx_version}"
+  if [ "${cxx_version}" != "${cxx_build_version}" ]; then
```
Comment (dtolnay) [inline .github/workflows/metadata-check.sh:32]: This is already enforced by https://github.com/dtolnay/cxx/blob/1.0.124/Cargo.toml#L34.

**Review by dtolnay — COMMENTED** [review]
Of the 4 cxx version checks in metadata-check.sh, the only 1 that matters is not implemented correctly. That one should be fixed and the other 3 should be removed.

File: .github/workflows/metadata-check.sh [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    echo "Found differing cxx-build version ${cxx_build_version}"
+    ok = false
+  fi
+  if [ "${cxx_version}" != "${cxxbridge_cmd_version}" ]; then
+    echo "Found differing cxxbridge-cmd version ${cxxbridge_cmd_version}"
+    ok = false
+  fi
```
Comment (dtolnay) [inline .github/workflows/metadata-check.sh:39]: I published cxx 1.0.133 which automatically puts a correct version of cxxbridge-cmd into the lockfile of any crate that depends on cxx (https://github.com/dtolnay/cxx/pull/1408). Could you try upgrading, reverting the taskchampion-cpp/Cargo.toml change from this PR, and see if cxxbridge-cmd stays in the lockfile? If so then this check can also be removed.

File: .github/workflows/metadata-check.sh (hunk 2 above)
Comment (djmitche) [reply .github/workflows/metadata-check.sh:26]: 🤦🏼

File: .github/workflows/metadata-check.sh (hunk 6 above)
Comment (djmitche) [reply .github/workflows/metadata-check.sh:39]: Thanks! I added https://github.com/GothenburgBitFactory/taskwarrior/pull/3714 to try this.

**Pull request merged**